modules/candidate_parameters/jsx/monitoring.js

```diff
@@ -0,0 +1,449 @@
+          lineInfo += "]";
+
+          line.push(
+            <td>
+              <b>
+                Visit->
+              </b>
+              {visitH}
+            </td>
+          );
+          line.push(
+            <td class="td-hist">
+              <b>
+                Flagged ->
+              </b>
+              {flaggedH}
+            </td>
+          );
+          line.push(
+            <td class="td-hist">
+              <b>
+                Completed monitoring ->
+              </b>
+              {monitoredH}
+            </td>
+          );
+          if (monitoredDateH !== null) {
+            line.push(
+              <td class="td-hist">
+                <b>
+                  Date of completion ->
+                </b>
+                {monitoredDateH}
+              </td>
+            );
+          }
+
+          if (examinerH !== null) {
+            line.push(
+              <td class="td-hist">
+                <b>
+                  Monitored by ->
+                </b>
+                {this.state.Data.examiners[examinerH]}
+              </td>
+            );
+          }
+
+          formattedHistory.push(
+            <tr>
+              <td style={{padding: "1px", width: "25%"}}>
+                <i>{lineInfo}</i>
+              </td>
+              {line}
+            </tr>
+            );
+        }
+      }
+
+      var alertMessage = "";
+      var alertClass = "alert text-center hide";
+      if (this.state.updateResult) {
+        if (this.state.updateResult === "success") {
+          alertClass = "alert alert-success text-center";
+          alertMessage = "Update Successful!";
+        } else if (this.state.updateResult === "error") {
+          var errorMessage = this.state.errorMessage;
+          alertClass = "alert alert-danger text-center";
+          alertMessage = errorMessage ? errorMessage : "Failed to update!";
+        }
+      }
+
+      return (
+        <div class="row">
+          <FormElement
+            name ="monitoringStatus"
+            onSubmit ={this.handleSubmit}
+            ref ="form"
+            class ="col-md-12"
+          >
+            <StaticElement
+              label ="PSCID"
+              text ={this.state.Data.pscid}
+            />
+            <StaticElement
+              label ="DCCID"
+              text ={this.state.Data.candID}
+            />
+            {visits}
+            <div className ={alertClass} role="alert" ref="alert-message">
+              {alertMessage}
+            </div>
+            {updateButton}
+            <table className="table-hist">
+              {formattedHistory}
+            </table>
+
+          </FormElement>
+        </div>
+      );
+    },
+    /**
+     * Handles form submission
+     *
+     * @param {event} e - Form submission event
+     */
+    handleSubmit: function(e) {
+      e.preventDefault();
+      var myFormData = this.state.formData;
+      var today = new Date();
+      var dd = today.getDate();
```

Comment:
Instead of all this hacking around you can do:
```js
const input = new Date(this.state.formData.dateMonitored);
const today = new Date():
```
Then compare the two date objects